.github/workflows/test/verify-run-status.test.js

```diff
@@ -0,0 +1,410 @@
+          githubMock,
+          createMockContext(),
+          createMockCore(),
+          "checkRunName",
+          "head_sha",
+        ),
+    ).rejects.toThrow(/Multiple completed check runs with name/);
+  });
+});
+
+describe("getWorkflowRun", () => {
+  it("returns matching workflow_run", async () => {
+    const githubMock = createMockGithub();
+    githubMock.rest.actions.listWorkflowRunsForRepo = vi
+      .fn()
+      .mockResolvedValue({
+        data: {
+          workflow_runs: [
+            {
+              name: "workflowName",
+              status: "completed",
+              conclusion: "success",
+            },
+          ],
+        },
+      });
+
+    const actual = await getWorkflowRun(
+      githubMock,
+      createMockContext(),
+      createMockCore(),
+      "workflowName",
+      "head_sha",
+    );
+
+    expect(actual).toEqual(
+      expect.objectContaining({
+        name: "workflowName",
+        status: "completed",
+        conclusion: "success",
+      }),
+    );
+  });
+
+  it("returns null when no workflow matches", async () => {
+    const githubMock = createMockGithub();
+    githubMock.rest.actions.listWorkflowRunsForRepo = vi
+      .fn()
+      .mockResolvedValue({
+        data: {
+          workflow_runs: [
+            {
+              name: "otherWorkflowName",
+            },
+          ],
+        },
+      });
+
+    const actual = await getWorkflowRun(
+      githubMock,
+      createMockContext(),
+      createMockCore(),
+      "workflowName",
+      "head_sha",
+    );
+
+    expect(actual).toBeNull();
+  });
+
+  it("returns latest when multiple workflows match", async () => {
+    const githubMock = createMockGithub();
+    const earlyDate = "2025-04-01T00:00:00Z";
+    const laterDate = "2025-04-02T00:00:00Z";
+    githubMock.rest.actions.listWorkflowRunsForRepo = vi
+      .fn()
+      .mockResolvedValue({
+        data: {
+          workflow_runs: [
+            {
+              name: "workflowName",
+              status: "completed",
+              conclusion: "success",
+              updated_at: earlyDate,
+            },
+            {
+              name: "workflowName",
+              status: "completed",
+              conclusion: "success",
+              updated_at: laterDate,
+            },
+          ],
+        },
+      });
+
+    const actual = await getWorkflowRun(
+      githubMock,
+      createMockContext(),
+      createMockCore(),
+      "workflowName",
+      "head_sha",
+    );
+
+    expect(actual).toEqual(
+      expect.objectContaining({
+        name: "workflowName",
+        status: "completed",
+        conclusion: "success",
+        updated_at: laterDate,
+      }),
+    );
+  });
+});
+
+describe("verifyRunStatus", () => {
+  beforeEach(() => {
+    delete process.env.CHECK_RUN_NAME;
+    delete process.env.WORKFLOW_NAME;
+  });
+
+  afterAll(() => {
+    delete process.env.CHECK_RUN_NAME;
+    delete process.env.WORKFLOW_NAME;
+  });
```

Comment:
To avoid this, create a helper function `verifyRunStatusImpl()` in `verify-run-status.js`, which accepts these parameters as a  string argument rather than reading from env.  Only read from env in CI, otherwise pass the parameters directly in tests.